custom_components/apple_tv/__init__.py

```diff
@@ -364,6 +365,12 @@ def address_updated(self, address):
         )
         self.hass.add_job(update_entry, self.config_entry)
 
+    @staticmethod
+    async def async_options_updated(hass, config_entry):
```

Comment:
@postlund can you assist, please? :)
I've added config_entry update listener and it's working as expected.
The only problem that I am not able to pass `self` to `async_options_updated` so I will be able to change `self._is_pwr_mgmt_on`.

Thanks in advance.

Reply:
Try removing @staticmethod an add self as first argument.

Reply:
getting following error:
```
  File "/usr/local/lib/python3.7/site-packages/aiohttp/web_protocol.py", line 418, in start,
    resp = await handler(request),
  File "/usr/local/lib/python3.7/site-packages/aiohttp/web_app.py", line 458, in _handle,
    resp = await task,
  File "/usr/local/lib/python3.7/site-packages/aiohttp/web_middlewares.py", line 119, in impl,
    return await handler(request),
  File "/usr/src/homeassistant/homeassistant/components/http/real_ip.py", line 39, in real_ip_middleware,
  File "/usr/src/homeassistant/homeassistant/components/http/ban.py", line 72, in ban_middleware,
  File "/usr/src/homeassistant/homeassistant/components/http/auth.py", line 127, in auth_middleware,
  File "/usr/src/homeassistant/homeassistant/components/config/config_entries.py", line 227, in post,
  File "/usr/src/homeassistant/homeassistant/components/http/view.py", line 123, in handle,
    result = await result,
    return await super().post(request, flow_id),
  File "/usr/src/homeassistant/homeassistant/components/http/data_validator.py", line 50, in wrapper,
  File "/usr/src/homeassistant/homeassistant/helpers/data_entry_flow.py", line 92, in post,
    result = await method(view, request, *args, **kwargs),
    result = await self._flow_mgr.async_configure(flow_id, data),
  File "/usr/src/homeassistant/homeassistant/data_entry_flow.py", line 130, in async_configure,
  File "/usr/src/homeassistant/homeassistant/data_entry_flow.py", line 198, in _async_handle_step,
    result = await self._async_handle_step(flow, cur_step["step_id"], user_input),
    result = await self.async_finish_flow(flow, dict(result)),
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 935, in async_finish_flow,
    self.hass.config_entries.async_update_entry(entry, options=result["data"]),
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 747, in async_update_entry,
    self.hass.async_create_task(listener(self.hass, entry)),
TypeError: 'NoneType' object is not callable,
2020-03-21 19:17:14 ERROR (MainThread) [aiohttp.server] Error handling request,
Traceback (most recent call last):
```

Reply:
Hmm, that makes it a bit more "complicated". Let me look into it and I'll get back to you.

Reply:
Ok, so, I believe we should do it like this. Store the value as for instance `self.is_pwr_mgmt_on` in the manager class. Then write a callback method similar to this:

```python
    @staticmethod
    async def async_options_updated(hass, entry):
        """Triggered by config entry options updates."""
        hass.data[DOMAIN][entry.entry_id].is_pwr_mgmt_on = entry.options[CONF_PWR_MGMT]
```

Should work.

Reply:
I've already tried a similar solution and I've got the following error:
```
2020-03-23 10:27:31 ERROR (MainThread) [homeassistant.core] Error doing job: Task exception was never retrieved,
Traceback (most recent call last):,
    hass.data[DOMAIN][entry.entry_id].is_pwr_mgmt_on = entry.options[CONF_PWR_MGMT],
  File "/config/custom_components/apple_tv/__init__.py", line 371, in async_options_updated,
KeyError: '2dc26103d96a4b37801a1221a0bc7ed0'
```

Reply:
Ah, right. It should be `entry.unique_id` and not `entry.entry_id`.

Reply:
@postlund yep, that one seems to work.
I will test it and push the fix.